File: html/cgi/php/ColorSetter.php
<?php
// Enable error reporting for debugging
error_reporting(E_ALL);
ini_set('display_errors', 1);

// Retrieve form data and server name
$form = $_POST;
$serverName = $_SERVER['SERVER_NAME'] ?? '';

// Initialize the HTML body
$body = "<!DOCTYPE html>
<html>
<head>
    <title>BackGround Color Is Modified</title>
    <style>
        * { font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif; }
        body { background-color: Azure; display: flex; justify-content: center; align-items: center; height: 100vh; margin: 0; flex-direction: column; }
        h1 { font-size: 3em; color: black; }
    </style>
</head>
<body>
    <h1>Color Successfully Modified</h1>
    <br><a href=\"/php/welcome.php\"><button>Go back to your profile</button></a>
</body>
</html>";

// Print the HTTP headers

// Set the background color cookie if it's in the form data
if (isset($form['BackGroundColor'])) {
    $bgcolor = htmlspecialchars($form['BackGroundColor']);
    setcookie('bgcolor', $bgcolor,  time() + (86400 * 30), '/'); // 1 hour expiration
}

header("Content-Type: text/html");
header("Content-Length: " . strlen($body));

// Print the server name and current date if available
if (!empty($serverName)) {
    echo "Server: $serverName<br>";
}

$date = gmdate("D, d M Y H:i:s T");
echo "Date: $date<br>";

// Print the HTML body
echo $body;
?>

File: html/cgi/perl/ColorSetter.pl
#!/usr/bin/perl
use strict;
use warnings;
use POSIX qw(strftime);

# Function to decode URL-encoded form data
sub uri_unescape {
    my $str = shift;
    $str =~ s/%([0-9A-Fa-f]{2})/chr(hex($1))/ge;
    return $str;
}

# Function to read form data from QUERY_STRING
sub read_form_data {
    my $query_string = $ENV{'QUERY_STRING'} || '';
    my %form;
    foreach my $pair (split /&/, $query_string) {
        my ($key, $value) = split /=/, $pair, 2; # Split only once to handle values with '='
        $key = uri_unescape($key);
        $value = uri_unescape($value);
        push @{$form{$key}}, $value;
    }
    return %form;
}

# Read form data
my %form = read_form_data();

# Default background color
my $bgcolor = "Azure";

# Update background color if provided in form data
if (exists $form{'BackGroundColor'}) {
    $bgcolor = $form{'BackGroundColor'}[0];
}

# Prepare the HTML body with dynamic background color
my $body = <<"END_HTML";
<!DOCTYPE html><html>
<head>
<meta charset="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BackGround Color Is Modified</title>
<style>
* {font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}
body { background-color: Azure; display: flex; justify-content: center; align-items: center; height: 100vh; margin: 0; flex-direction: column;}
h1 { font-size: 3em; color: black; }
</style></head>
<body><h1>Color Successfully Modified</h1></body></html>
<br><a href="/perl/welcome.pl"><button> Go back to your profile</button></a></html>
END_HTML

# Print HTTP headers
print "Content-Type: text/html\r\n";
print "Cache-Control: no-cache\r\n";
print "Content-Length: " . length($body) . "\r\n";

# Set cookies if form data is present
if (exists $form{'BackGroundColor'}) {
    my $bgcolor = $form{'BackGroundColor'}[0];
    print "Set-Cookie: bgcolor=$bgcolor\r\n";
}

# Print server name and date if available
if (exists $ENV{'SERVER_NAME'}) {
    my $server_name = $ENV{'SERVER_NAME'};
    print "Server: $server_name\r\n";
}

my $date = strftime "%a, %d %b %Y %H:%M:%S %Z", localtime;
print "Date: $date\r\n";

# Print the HTML body
print "\r\n";  # Ensure a blank line before the body
print $body;
